### DISCUSSION TITLE
RTX Video Super Resolution 英伟达视频超分辨率 VSR & RTX HDR

### DISCUSSION THREAD

**Discussion by Xueipyet** [hooke007/mpv_PlayKit#286] [mpv 通用]
MPV能调用它吗？

**Comment by hooke007** [comment]
## RTX VSR
使用说明迁移至 https://github.com/hooke007/mpv_PlayKit/discussions/607#discussioncomment-13940167
本帖仅留作讨论反馈专用

**Comment by hooke007** [comment]
## RTX VSR
使用说明迁移至 https://github.com/hooke007/mpv_PlayKit/discussions/607#discussioncomment-13940203
本帖仅留作讨论反馈专用

**Reply by hooke007** [reply]
<s>

## 过时的解决方案4（已失效）：
根据问题中的评论做的快速测试。
下载 nvdia app https://www.nvidia.cn/software/nvidia-app/
手动添加 mpv.exe 并关闭&启用对应的功能

<img width=200 src="https://github.com/hooke007/MPV_lazy/assets/41094733/e2b4dac4-0b7c-445d-9ea2-a7e17c1fc78c">

**(如果上面那个选项灰的那么应该修改Win11的下两项设置)**

<img width=200 src="https://github.com/hooke007/MPV_lazy/assets/41094733/eb974f5d-7556-49fd-8de2-aea989276c3f">

...................
mpv的全默认参数即可激活rtx hdr，
但是修改成 `vo=gpu-next` 就无法激活，尚未调查原因。

</s>

**Comment by zhongfly** [comment]
https://github.com/mpv-player/mpv/commit/2848af5618fa823571cf4ec8cc2a4580d37f1648
超分辨率的功能已经合并
使用选项：`--vf=d3d11vpp=scaling-mode=nvidia`
我的构建已经有更新了 https://github.com/zhongfly/mpv-winbuild/releases/latest

**Reply by signin404** [reply]
没效果 我哪里做错了么?

![QQ截图20240820131757](https://github.com/user-attachments/assets/d9f5b620-7166-4adc-bfe6-cf56e67d2795)

**Reply by hooke007** [reply]
无效信息。
去看控制台或者日志

**Reply by hooke007** [reply]
d3d11vpp按理说也需要d3d11的全链路，你确定nvdec能行？

**Reply by signin404** [reply]
> d3d11vpp按理说也需要d3d11的全链路，你确定nvdec能行？

我试了软解也不行

![QQ截图20240820134326](https://github.com/user-attachments/assets/f5f620fc-fefe-4f0f-b7b9-2958c8f49cf0)
![QQ截图20240820134354](https://github.com/user-attachments/assets/0ae696d9-00dd-4019-8e4d-39798cb0f29a)

**Reply by hooke007** [reply]
输入格式错误

**Reply by hooke007** [reply]
我粗看了一眼代码，好像只支持nv12，所以第一步你把脚本啥的配置里乱起八糟的东西都拿掉，只剩一个hwdec=d3d11va试试

**Reply by signin404** [reply]
> 我粗看了一眼代码，好像只支持nv12，所以第一步你把脚本啥的配置里乱起八糟的东西都拿掉，只剩一个hwdec=d3d11va试试

还是不行

![QQ截图20240820141819](https://github.com/user-attachments/assets/28d9aa40-2bf0-4af5-98aa-9a62012a7e3c)
![QQ截图20240820142024](https://github.com/user-attachments/assets/6df39037-6db6-4247-87eb-315588926fe7)

**Reply by hooke007** [reply]
控制台里没显示东西，看日志

**Reply by signin404** [reply]
> 控制台里没显示东西，看日志

[mpv-lazy.log](https://github.com/user-attachments/files/16670420/mpv-lazy.log)

**Reply by hooke007** [reply]
你是不是要设置一下缩放参数。。。
![image](https://github.com/user-attachments/assets/53a14e23-9ceb-4d99-8f1a-a3e895b01403)

**Reply by WilliamK7** [reply]
我这边有同样问题，日志中关键部分如下：
```
[   0.371][d][ffmpeg/video] h264: Reinit context to 1280x720, pix_fmt: yuv420p
[   0.371][d][vd] DR parameter change to 1280x738 yuv420p align=64
[   0.371][d][vd] Allocating new (host-cached) DR image...
[   0.371][d][vd] ...failed..
[   0.371][v][vd] DR failed - disabling.
[   0.384][v][vd] Using software decoding.
[   0.384][v][vd] Decoder format: 1280x720 [0:1] yuv420p bt.709/bt.709/bt.1886/limited/auto CL=mpeg2/4/h264 crop=1280x720+0+0
[   0.384][v][vf] [in] 1280x720 yuv420p bt.709/bt.709/bt.1886/limited/display CL=mpeg2/4/h264 crop=1280x720+0+0
[   0.384][v][vf] [userdeint] 1280x720 yuv420p bt.709/bt.709/bt.1886/limited/display CL=mpeg2/4/h264 crop=1280x720+0+0
[   0.384][v][vf] [userdeint] (disabled)
[   0.384][v][vf] [d3d11vpp] 1280x720 yuv420p bt.709/bt.709/bt.1886/limited/display CL=mpeg2/4/h264 crop=1280x720+0+0
[   0.384][v][vo/gpu-next] Loading hwdec drivers for format: 'd3d11'
[   0.384][v][vo/gpu-next] Loading hwdec driver 'd3d11-egl'
[   0.384][v][vo/gpu-next] Loading failed.
[   0.384][d][hwupload] looking at format d3d11/nv12
[   0.384][d][hwupload]   supports: nv12
[   0.384][d][hwupload] looking at format d3d11/p010
[   0.384][d][hwupload]   supports: p010
[   0.384][d][hwupload] looking at format d3d11/bgra
[   0.384][d][hwupload]   supports:
[   0.384][d][hwupload]   not supported by VO: bgra
[   0.384][d][hwupload] looking at format d3d11/x2bgr10
[   0.384][d][hwupload]   supports:
[   0.384][d][hwupload]   not supported by VO: x2bgr10
[   0.384][d][hwupload] looking at format d3d11/rgbaf16
[   0.384][d][hwupload]   supports:
[   0.384][d][hwupload]   not supported by VO: rgbaf16
[   0.384][d][hwupload] looking at format d3d11/vuyx
[   0.384][d][hwupload]   supports:
[   0.384][d][hwupload]   not supported by VO: vuyx
[   0.384][d][hwupload] looking at format d3d11/yuyv422
[   0.384][d][hwupload]   supports:
[   0.384][d][hwupload]   not supported by VO: yuyv422
[   0.384][d][hwupload] looking at format d3d11/y210
[   0.384][d][hwupload]   supports:
[   0.384][d][hwupload]   not supported by VO: y210
[   0.384][d][hwupload] looking at format d3d11/xv30
[   0.384][d][hwupload]   supports:
[   0.384][d][hwupload]   not supported by VO: xv30
[   0.384][d][hwupload] looking at format d3d11/y212
[   0.384][d][hwupload]   supports:
[   0.384][d][hwupload]   not supported by VO: y212
[   0.384][d][hwupload] looking at format d3d11/xv36
[   0.384][d][hwupload]   supports:
[   0.384][d][hwupload]   not supported by VO: xv36
[   0.384][d][hwupload] looking at format d3d11/yuv420p
[   0.384][d][hwupload]   supports:
[   0.384][e][hwupload] Unable to find a compatible upload format for yuv420p
[   0.384][e][autoconvert] Failed to create HW uploader for format yuv420p
[   0.384][e][autoconvert] can't find video conversion for yuv420p
[   0.384][e][vf] Disabling filter d3d11vpp.00 because it has failed.
```

**Reply by signin404** [reply]
> 你是不是要设置一下缩放参数。。。 

设置了也不行
vf=d3d11vpp=scaling-mode=nvidia:scale=1.0

**Reply by hooke007** [reply]
1.0就是默认值，意思是不缩放

**Reply by signin404** [reply]
> 1.0就是默认值，意思是不缩放

改成2.0可以了
这竟然不是自动的…

**Reply by WilliamK7** [reply]
我这里设置了缩放参数还是不行，控制台会报错：
```
[hwupload] Unable to find a compatible upload format for yuv420p
[autoconvert] Failed to create HW uploader for format yuv420p
[autoconvert] can't find video conversion for yuv420p
[vf] Disabling filter d3d11vpp.00 because it has failed.
```
和前面日志里是一样的

**Reply by signin404** [reply]
> 我这里设置了缩放参数还是不行，控制台会报错：

软解不行
要用硬解d3d11va或nvdec
-copy可以

**Reply by zhongfly** [reply]
https://github.com/mpv-player/mpv/pull/14155#issuecomment-2162495462
可能是yuv420p不能转换成nv12格式的问题，原pr里有说到这个限制，没想到还是没解决？

建议换个其他格式的视频试试…

**Reply by WilliamK7** [reply]
从软解换成 `hwdec=auto-copy` 就可以了，就是控制台里会有不影响观看的警告：
```
[d3d11vpp]No fitting video processor found, picking #0
[autoconvert]unexpected AVFrame/imgfmt hardware context mismatch
```

**Reply by hooke007** [reply]
在没有必要需求前提下不要copy，目前来看不接受软解出来的格式。
指定了copy之后还会多做一步转换浪费性能

**Reply by hooke007** [reply]
> > 1.0就是默认值，意思是不缩放
> 
> 改成2.0可以了
> 这竟然不是自动的…

当然不是自动的，vpp的缩放本身就支持不同倍率，这个pr只是扩展了支持的缩放模式

**Reply by WilliamK7** [reply]
>在没有必要需求前提下不要copy，目前来看不接受软解出来的格式。

ok，了解了，主要是看 MPV_lazy 中给的选项默认就是 auto-copy 就用了

**Reply by hooke007** [reply]
因为大部分滤镜是cpu滤镜，不支持native hwdec
这是个综合考量下的妥协设定

**Reply by signin404** [reply]
> 当然不是自动的，vpp的缩放本身就支持不同倍率，这个pr只是扩展了支持的缩放模式

有自动计算窗口大小参数的话 能更方便跟其他模型混用
而且就算单独用 不同分辨率片源还得分别写 不友好…
有打算添加进K7sfunc里么?

**Reply by hooke007** [reply]
自动计算可以脚本实现。简易的计算甚至可以只靠profile实现
这个不是vs滤镜，写不进k7f里

**Reply by signin404** [reply]
不兼容其他模型
报跟软解一样的错
测试了DPIR OpenProteus RIFE

**Reply by hooke007** [reply]
这玩意和vs滤镜都不是一条路的东西，，谈不上兼容不兼容
vs滤镜输入和输出都是yuv（和你在vs里怎么折腾都无关），这需要手动改滤镜链排列

**Reply by signin404** [reply]
> 这玩意和vs滤镜都不是一条路的东西，，谈不上兼容不兼容 vs滤镜输入和输出都是yuv（和你在vs里怎么折腾都无关），这需要手动改滤镜链排列

好像不对 报错不同
VSR有效 VS无效
始终是顺序在前的才有效
![QQ截图20240820182255](https://github.com/user-attachments/assets/d5ae1a6e-21b6-42ef-a221-08e960a0a51d)

**Reply by hooke007** [reply]
> 这需要手动改滤镜链排列

你没看懂这句话的意思。滤镜之间的输入输出格式不一样。需要手动插入格式转换。autoconvert很多情况无法检测滤镜之间需要的格式

**Reply by signin404** [reply]
> > 这需要手动改滤镜链排列
> 
> 你没看懂这句话的意思。滤镜之间的输入输出格式不一样。需要手动插入格式转换。autoconvert很多情况无法检测滤镜之间需要的格式

确实可以了
clip = k7f.FMT_CHANGE(clip, fmtc=False, fmt_pix=2, dither=1)

**Reply by hooke007** [reply]
。。。你还是没懂，不过看上去瞎猫打到死耗子了

**Comment by AcerolaShinobu** [comment]
测试10bit片源无法调用RTX VSR，即使是d3d11va硬解。
![图片](https://github.com/user-attachments/assets/63efc8e7-4ff0-43f9-adc8-0d9641567c8d)

**Reply by AcerolaShinobu** [reply]
修改格式为nv12可以正常运行，
local format = mp.get_property_native("video-params/hw-pixelformat") == "nv12" and "" or "format=nv12,"

**Reply by hooke007** [reply]
你的版本不对，这边不存在这个问题，上日志
你贴的这行莫名其妙的代码的意思是片源是8bit，我看不出来和片源是10bit有啥关联

**Reply by AcerolaShinobu** [reply]
[mpv-lazy.log](https://github.com/user-attachments/files/16791981/mpv-lazy.log)

日志用的最新版整合包。
代码是方便观看10bit视频的脚本，遇到非nv12格式的修改为nv12。

**Reply by hooke007** [reply]
你上传的日志是我要的10bit不能工作的吗，你的日志里显示正常放大

**Reply by AcerolaShinobu** [reply]
NVIDIA控制面板 内显示 RTX VSR 非活动，任务管理器 GPU 里的 Compute_0 和 Cuda 占用率为 0

**Reply by hooke007** [reply]
这个错误没人报过，因为p010本来就是不需要转换的
或者你给我能复现的样本

**Reply by AcerolaShinobu** [reply]
我这 MPV 所有10bit 的视频都不能工作，MPC Video Renderer 就正常

**Reply by AcerolaShinobu** [reply]
https://github.com/user-attachments/assets/a73192e3-f079-44c8-b741-3167efe43185

找到个10M内的mkv，虽然是静止画面

**Reply by AcerolaShinobu** [reply]
换回[mpv-x86_64-v3-20240826-git-cb4fdb5](https://github.com/shinchiro/mpv-winbuild-cmake/releases/download/20240826/mpv-x86_64-v3-20240826-git-cb4fdb5.7z)，10bit片源可以调用 RTX VSR，也许是新版本有问题

**Reply by hooke007** [reply]
已复现并汇报

**Reply by hooke007** [reply]
已修复

**Comment by wz89989123** [comment]
大佬，RTX HDR用NVIDIA APP的那个方法好像没用，不知道是不是app更新的原因，不支持的软件就不显示自动rtx的滤镜选项了，mpv里调用rtx hdr现在无法实现了吗

**Reply by hooke007** [reply]
我也无法让它实际工作

**Reply by qqqs4ss** [reply]
> 我也无法让它实际工作

可以用这个mod给mpv开RTXHDR,本来是给游戏用的，但是mpv也可以用，我一直这么用的。
需要先在nv控制面板里管理3d设置-程序设置，把mpv添加进去
https://www.nexusmods.com/site/mods/781

**Reply by 1354733788** [reply]
> > 我也无法让它实际工作
> 
> 可以用这个mod给mpv开RTXHDR，本来是给游戏用的，但是mpv也可以用，我一直这么用的。 需要先在nv控制面板里管理3d设置-程序设置，把mpv添加进去 https://www.nexusmods.com/site/mods/781

你好大佬我按你说的把mpv添加进了程序设置里了，显示器hdr也开了但是这个软件不起作用，能教教我具体的操作吗谢谢

**Reply by qqqs4ss** [reply]
> > > 我也无法让它实际工作
> > 
> > 
> > 可以用这个mod给mpv开RTXHDR，本来是给游戏用的，但是mpv也可以用，我一直这么用的。 需要先在nv控制面板里管理3d设置-程序设置，把mpv添加进去 https://www.nexusmods.com/site/mods/781
> 
> 你好大佬我按你说的把mpv添加进了程序设置里了，显示器hdr也开了但是这个软件不起作用，能教教我具体的操作吗谢谢

先打开NvTrueHdr,把mpv.exe拖进nvtruehdr的窗口回车，然后写入veryhigh、hud那几个选项中的一个回车就可以了。可以先用hud，mpv播放时左上角如果显示2个灰色方块表示已经启用了

**Reply by hooke007** [reply]
有灰方但是并没有看到面板里的活动指标。目视效果也与edge相差甚远

**Reply by qqqs4ss** [reply]
> 有灰方但是并没有看到面板里的活动指标。目视效果也与edge相差甚远

面板的活动指标是什么？你可以试试veryhigh，我这边目视和edge差不太多

**Reply by hooke007** [reply]
n卡的控制面板会显示rtx hdr vsr的活动状态

**Reply by qqqs4ss** [reply]
> n卡的控制面板会显示rtx hdr vsr的活动状态

我这边播放时控制面板也是显示非活动，但是效果很明显，有白色的画面暂停等2秒白色部分明显变暗就是原来sdr的效果，再恢复播放明显增亮。
mod下载页面有个TrueHDRTweaks 0.6可以详细调整峰值亮度、饱和度、gamma之类的参数

**Reply by hooke007** [reply]
暂停状态确实是能看出hdr介入的，恢复播放亮度就拉平了

**Reply by qqqs4ss** [reply]
> 暂停状态确实是能看出hdr介入的，恢复播放亮度就拉平了

亮度垃平意思是高亮部分又变暗了么？你是不是mpv装了一些其它东西或设置播放时把hdr效果覆盖掉了，我这边就用的最新24v2版本体就能播放时高亮部分hdr增亮，窗口播放且暂停时会变暗。你重新纯净安装一遍mpv试试？

**Reply by hooke007** [reply]
所有测试当然是在no-config下操作的

**Reply by qqqs4ss** [reply]
> 所有测试当然是在no-config下操作的

试试关win11 autohdr。nv控制面板mpv的  "Vulkan/OpenGL present method" setting in NVCP changed to "Prefer layered on DXGI swapchain" for HDR to activate properly.
我这没有你那个情况，不知道哪的问题，建议纯净安装用其它hdr选项试试。你电脑里有dx10 11 12游戏的话可以试试对游戏有没有效果

**Reply by hooke007** [reply]
能想到的早试过了。最好的结果就是前面说的能出hdr，但效果差edge很多。我的显示器是oled，这个目视效果怎么自我欺骗都达不到类似的效果。

**Reply by qqqs3ss** [reply]
> 暂停状态确实是能看出hdr介入的，恢复播放亮度就拉平了

我好像知道是怎么回事了，我今天更新v3也出现了这种情况，画面灰白平还不如sdr。你当时是不是更新了mpv版本，或者转换了bt2020色域？发现新版本mpv v3好像是会默认把sdr转换成bt2020色域的hdr视频播放？而用NvTrueHdr强制给bt2020色域的视频再加rtx hdr效果就会出现这种情况。

v3版本，好像并不能正确启用rtx hdr，我在mpvconfig里加入了 
hwdec = d3d11va
d3d11-output-csp = srgb                     # RTX HDR 应该只对输出SDR内容的窗口生效，因此尝试锁定输出色域
vf-append = d3d11vpp=format=nv12:nvidia-true-hdr
这3条，整体画面亮度变暗了，但是好像亮度那一层完全没加上去，该亮的没亮起来。

请问有没有方法避免mpv把sdr视频自动转换成bt2020色域？v3我强制rtx hdr方法失效、自带的又不生效，这样我只能停留在v2了。

**Reply by hooke007** [reply]
这都是什么根什么，说“暂停状态确实是能看出hdr介入的，恢复播放亮度就拉平了”的时候mpv还没实现rtx hdr。我现在不需要外部软件实现。

> 默认把sdr转换成bt2020

之前某个版本的默认值变auto了，这个检测机制一般不准确，强制关掉它
https://mpv.io/manual/master/#options-target-colorspace-hint

**Reply by qqqs3ss** [reply]
> 这都是什么根什么，说“暂停状态确实是能看出hdr介入的，恢复播放亮度就拉平了”的时候mpv还没实现rtx hdr。我现在不需要外部软件实现。
> 
> > 默认把sdr转换成bt2020
> 
> 之前某个版本的默认值变auto了，这个检测机制一般不准确，强制关掉它 https://mpv.io/manual/master/#options-target-colorspace-hint

请问这行HDR10:diplay 能不能去掉？新版只要打开win11hdr，不管播放sdr还是hdr都会出现这行，老版本只有播放hdr视频时会出现，只要出现这行，我强制加rtx hdr就会发灰。display那后面好像是亮度信息？我这里永远显示0.02 不知道对不对.
ps 你引用我的那句没截全
<img width=300 src=https://github.com/user-attachments/assets/abe1f88f-6926-4abc-8c5c-4df1fcb28142> <img width=300 src=https://github.com/user-attachments/assets/f21c26b1-5901-492d-ba27-a4d397866e26>

**Reply by hooke007** [reply]
加一条 `target-peak=203` 试试，中间隔了两个月我还没研究mpv改了什么。

**Reply by qqqs3ss** [reply]
> 加一条 `target-peak=203` 试试，中间隔了两个月我还没研究mpv改了什么。

可以了，非常感谢！不知道这是什么原理
发现加上这条播放hdr原生视频会变暗，播放原生hdr时还得把它去掉

**Comment by tanyiok1234** [comment]
mpv内部调用RTX video HDR虽然日志显示成功，但画面没有效果（相比浏览器触发的效果来说）。NV app也没有指示。各种组合都试了一边，没能成功。

另一方面，NV的10月驱动支持了RTX HDR在多屏上的启用，开启以后不需要什么特别的操作就可以直接触发，肉眼可见的区别，和浏览器播放效果比较一致。
（但是要打开NV那个滤镜设置一下中间灰的亮度）

**Comment by YiyangLu** [comment]
mpv 调用RTX super resolution, 播放一小时后 会大概率引发 蓝屏.
根据minidump:
An attempt was made to access a pageable (or completely invalid) address at an
interrupt request level (IRQL) that is too high.  This is usually
caused by drivers using improper addresses.
最后debug是停在 nvlddmkm.sys 这个module..

不负责乱猜, 可能是:NULL地址引用

**Comment by wathper** [comment]
```
n                  vf toggle d3d11vpp=format=nv12:scale=2:scaling-mode=nvidia
m                  script-message cycle-commands  'change-list vf    pre "vapoursynth=~~/test.vpy,format=nv12,d3d11vpp=format=nv12:scale=2:scaling-mode=nvidia"'   'vf remove vapoursynth="~~/test.vpy" ; vf remove format=nv12 ; vf remove d3d11vpp=format=nv12:scale=2:scaling-mode=nvidia'
```
动态一键开关滤镜链的快捷键方案，需要 script-messages.lua 脚本。

**Comment by 1PercentSync** [comment]
放大是否始终在放大为2倍的情况下能达到最佳的画面质量，抑或是通过lua去匹配显示分辨率更好？
这里有一个实现匹配分辨率的lua脚本
https://github.com/mpv-player/mpv/pull/14698#issuecomment-2298643736

**Reply by hooke007** [reply]
我没有确认过它能实际支持放大几倍

**Reply by pUnivar** [reply]
对你提到的lua脚本进行了改进，能根据实际渲染大小去匹配scale而不是只匹配显示器分辨率，同时不影响其它vf
```
local last_scale = 1
local vsr_label = "@vsr"
local has_label = false

local function build_vsr_filter(scale, pixfmt)
    if pixfmt == "nv12" then
        return vsr_label..":d3d11vpp:scaling-mode=nvidia:scale="..scale
    elseif pixfmt == "yuv420p" then
        return vsr_label..":format=nv12,format=d3d11,d3d11vpp:scaling-mode=nvidia:scale="..scale
    end
    return nil
end

local function remove_vsr()
    if has_label == true then
        mp.command("vf remove ".. vsr_label)
        print("remove "..vsr_label)
        has_label = false
    end
end

local function set_scale(scale)
    print("[Scale] scale = "..scale)
    local pixfmt = mp.get_property_native("video-params/hw-pixelformat")
                or mp.get_property_native("video-params/pixelformat")
    local filter = build_vsr_filter(scale, pixfmt)
    if filter then
        print("vf append "..vsr_label)
        mp.command("vf append "..filter)
        has_label = true
    end
end

function Autovsr(name)
    local dst_width = mp.get_property_native("osd-dimensions/w")
    local dst_height = mp.get_property_native("osd-dimensions/h")
    local src_width = mp.get_property_native("width")
    local src_height = mp.get_property_native("height")
    if dst_width and dst_height and src_width and src_height then
        print(name.." @Dst: "..dst_width.."x"..dst_height.." @Src: "..src_width.."x"..src_height)
        local scale = math.min(dst_height / src_height, dst_width/ src_width)
        scale = math.floor(scale * 10 + 0.5) / 10
        scale = math.min(4.0, scale)
        if math.abs(scale - last_scale) < 0.05 then
            return
        else
            remove_vsr()
            last_scale = scale
            if scale > 1 then
                set_scale(scale)
            end
        end
    end
end

mp.observe_property("osd-dimensions/h", "native", Autovsr)
mp.observe_property("osd-dimensions/w", "native", Autovsr)
```

**Reply by hooke007** [reply]
这个原始脚本怎么说呢，ai 放大这种东西都是固定放大 2x 4x，虽然 nv 没有公开记录，但大概率面对这种不到 2x 的情况，内部2x 之后再用一个廉价的算法拉回来，这个廉价的步骤直接让 mpv 做也可以。

另外这个原始脚本做的时候 pr 刚开，所以他的格式转换做法也是比较 hack，现在都支持了 vpp 内部转换了你为啥还继承这个操作。
而且他完全没考虑p10 的支持。

**Reply by Akariln** [reply]
那么请问想改成根据分辨率自动单纯只放大2x或4x，然后让 mpv 自己拉回去，要怎么改呢？

看到的一些关联脚本 [1](https://gist.github.com/azumukupoe/a56a78d71cf26a8f682dc06407e615db) 、[2](https://github.com/mpv-player/mpv/discussions/14804#discussioncomment-12466574)

**Reply by hooke007** [reply]
这个不还是层主提到的脚本吗。

> 要怎么改呢

什么都不需要改啊，我二楼写的不就是吗？mpv默认最终输出缩限到窗口尺寸

**Reply by hooke007** [reply]
我没有验证过4x实际存不存在，根据源尺寸做不同profile可以尝试区分

**Reply by 1PercentSync** [reply]
> 我没有验证过4x实际存不存在，根据源尺寸做不同profile可以尝试区分

印象中是最高到2倍，来源已经不记得了，大概之前爬楼issue和pr看到的，高于2倍是在2倍基础上缩放

**Reply by hooke007** [reply]
验证起来也不难，找一个540p的视频分别拉到1080p和2160p，看cuda占用有没有爆发式的差异

**Reply by hooke007** [reply]
二层更新了两个简易的profile示例来处理源分辨率越界的问题，比脚本方案的区别就是控制台重复信息会多点，而且始终假定没有用别的滤镜预缩放处理分辨率。

**Reply by Akariln** [reply]
懂了，固定两倍就够了，谢谢各位

**Comment by Es4c** [comment]
![2025-02-09_204731](https://github.com/user-attachments/assets/d98b85ec-eee4-46ae-a6f1-f0177fd2e0d2)
按照二楼设置了一下，但是控制台会报参数错误，这个要怎么解决？

**Reply by hooke007** [reply]
没有遇到过

**Reply by zhongfly** [reply]
这个错误居然是中文的，有点奇怪？会不会和驱动版本有关

**Comment by lasxmidnight78** [comment]
为什么我添加
vf-append=format=nv12
vf-append=d3d11vpp=scale=2:scaling-mode=nvidia 
能开VSR，但是同时打开MVTOOL或者riff之后VSR就消失了

**Reply by hooke007** [reply]
你是按照我的操作来的吗？
![image](https://github.com/user-attachments/assets/3bdecdc8-9182-4b13-b5c7-a913bb488f30)

**Reply by JohNnimOn** [reply]
> 你是按照我的操作来的吗？ [#286 (comment)](https://github.com/hooke007/MPV_lazy/discussions/286#discussioncomment-5252792) ![image](https://private-user-images.githubusercontent.com/41094733/447301643-3bdecdc8-9182-4b13-b5c7-a913bb488f30.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTE4Nzc1NDQsIm5iZiI6MTc1MTg3NzI0NCwicGF0aCI6Ii80MTA5NDczMy80NDczMDE2NDMtM2JkZWNkYzgtOTE4Mi00YjEzLWI1YzctYTkxM2JiNDg4ZjMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzA3VDA4MzQwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEyOTg0OGUwNjEyOTVlYmNiMDg4Njk3MTgwOWZiMzgzZTcwYTczYTgxNzhiM2Q3ZjE4ODE3MTM1YWUxZDIxYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jmv59gCAbL5VpgHlBf-jBvqZ10pFNFuWWmlrT-PjfDM)

![屏幕截图 2025-07-07 163504](https://github.com/user-attachments/assets/3c6bf784-924a-45ac-a041-ea5182061b31)
我就是按照你那样设定的，也遇到这个问题

**Reply by hooke007** [reply]
你test的脚本内容是什么

**Reply by JohNnimOn** [reply]
> 你test的脚本内容是什么

我直接复制的忘了自己写具体用哪个滤镜，现在搞好了。还有我的显卡是5070 12G用rife_rtx有的视频能补有的视频会报错，选哪个滤镜好点呢

**Reply by hooke007** [reply]
请去版本贴反馈无关的问题。报错又是具体报什么错，请详细描述问题而不是让我没有头绪的乱猜。

**Comment by hooke007** [comment]
https://github.com/hooke007/mpv_PlayKit/discussions/286#discussioncomment-10498236
我发现我弄混了两个概念，RTX HDR 和 rtx video hdr 是两个东西，RTX HDR可以通过第三方工具强制开启，rtx video hdr始终未正常工作过。。。
